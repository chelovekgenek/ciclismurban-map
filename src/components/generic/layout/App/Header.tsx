import React from "react"
import { connect } from "react-redux"
import { Link } from "react-router-dom"

import { Divider, Dropdown, Menu, Icon } from "components/generic/ui"

import * as Styled from "./Header.style"
import { getAuthenticated, getEmail, LogoutAction } from "store/entities/user"
import { TAppState } from "store/entities"

interface IDispatchProps {
  logout: typeof LogoutAction
}
interface IStateProps {
  authenticated: ReturnType<typeof getAuthenticated>
  email: ReturnType<typeof getEmail>
}

interface IProps extends IStateProps, IDispatchProps {}

export const Header: React.FC<IProps> = ({ authenticated, email, logout }) => (
  <Styled.Container>
    <Link to="/">
      <Styled.Logo />
    </Link>
    <Styled.SectionRight>
      {authenticated ? (
        <Dropdown
          overlay={
            <Menu>
              <Menu.Item onClick={logout}>
                <Icon type="user" />
                Выйти
              </Menu.Item>
            </Menu>
          }
        >
          <Styled.Profile>
            {email} <Icon type="down" />
          </Styled.Profile>
        </Dropdown>
      ) : (
        <React.Fragment>
          <Link to="/login">Войти</Link>
          <Divider type="vertical" />
          <Link to="/register">Зарегестрироваться</Link>
        </React.Fragment>
      )}
    </Styled.SectionRight>
  </Styled.Container>
)

export default connect<IStateProps, IDispatchProps, null, TAppState>(
  state => ({
    authenticated: getAuthenticated(state),
    email: getEmail(state),
  }),
  {
    logout: LogoutAction,
  },
)(Header)
