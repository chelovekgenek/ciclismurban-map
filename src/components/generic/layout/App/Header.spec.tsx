import React from "react"
import { shallow } from "enzyme"

import { Header } from "./Header"

describe("Header", () => {
  const render = (props: Partial<React.ComponentProps<typeof Header>> = {}) =>
    shallow(<Header authenticated={false} email={undefined} avatar={undefined} logout={jest.fn()} {...props} />)
  it("should match snapshot if not authenticated", () => {
    expect(render()).toMatchSnapshot()
  })
  it("should match snapshot if authenticated", () => {
    expect(render({ authenticated: true, email: "test@example.com", avatar: "link-to-pic" })).toMatchSnapshot()
  })
})
