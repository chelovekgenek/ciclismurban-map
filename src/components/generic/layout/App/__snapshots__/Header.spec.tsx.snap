// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot if authenticated 1`] = `
<Styled(Adapter)>
  <Link
    to="/"
  >
    <Styled(Component) />
  </Link>
  <styled.div>
    <Avatar
      src="link-to-pic"
    />
    <Dropdown
      mouseEnterDelay={0.15}
      mouseLeaveDelay={0.1}
      overlay={
        <Menu>
          <ForwardRef(Styled(MenuItem))>
            <ForwardRef
              to="/profile"
            >
              <Icon
                type="user"
              />
              Профиль
            </ForwardRef>
          </ForwardRef(Styled(MenuItem))>
          <ForwardRef(Styled(MenuItem))
            onClick={[MockFunction]}
          >
            <Icon
              type="logout"
            />
            Выйти
          </ForwardRef(Styled(MenuItem))>
        </Menu>
      }
      placement="bottomLeft"
    >
      <styled.span>
        test@example.com
         
        <Icon
          type="down"
        />
      </styled.span>
    </Dropdown>
  </styled.div>
</Styled(Adapter)>
`;

exports[`Header should match snapshot if not authenticated 1`] = `
<Styled(Adapter)>
  <Link
    to="/"
  >
    <Styled(Component) />
  </Link>
  <styled.div>
    <Link
      to="/login"
    >
      Войти
    </Link>
    <Divider
      type="vertical"
    />
    <Link
      to="/register"
    >
      Зарегестрироваться
    </Link>
  </styled.div>
</Styled(Adapter)>
`;
